cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/KafkaWriteOnlySampleStore.java

```diff
@@ -0,0 +1,275 @@
+ *   <li>{@link #MIN_BROKER_SAMPLE_STORE_TOPIC_RETENTION_TIME_MS_CONFIG}: The config for the minimal retention time for Kafka broker sample store
+ *   topic, default value is set to {@link #DEFAULT_MIN_BROKER_SAMPLE_STORE_TOPIC_RETENTION_TIME_MS}.</li>
+ * </ul>
+ */
+public class KafkaWriteOnlySampleStore implements SampleStore {
+  private static final Logger LOG = LoggerFactory.getLogger(KafkaSampleStore.class);
+  protected static final Duration PRODUCER_CLOSE_TIMEOUT = Duration.ofMinutes(3);
+  // Keep additional windows in case some of the windows do not have enough samples.
+  protected static final int ADDITIONAL_WINDOW_TO_RETAIN_FACTOR = 2;
+
+  protected static final short DEFAULT_SAMPLE_STORE_TOPIC_REPLICATION_FACTOR = 2;
+  protected static final int DEFAULT_PARTITION_SAMPLE_STORE_TOPIC_PARTITION_COUNT = 32;
+  protected static final int DEFAULT_BROKER_SAMPLE_STORE_TOPIC_PARTITION_COUNT = 32;
+  protected static final long DEFAULT_MIN_PARTITION_SAMPLE_STORE_TOPIC_RETENTION_TIME_MS = TimeUnit.HOURS.toMillis(1);
+  protected static final long DEFAULT_MIN_BROKER_SAMPLE_STORE_TOPIC_RETENTION_TIME_MS = TimeUnit.HOURS.toMillis(1);
+  protected static final String PRODUCER_CLIENT_ID = "KafkaCruiseControlSampleStoreProducer";
+
+  protected String _partitionMetricSampleStoreTopic;
+  protected String _brokerMetricSampleStoreTopic;
+  protected Short _sampleStoreTopicReplicationFactor;
+  protected int _partitionSampleStoreTopicPartitionCount;
+  protected int _brokerSampleStoreTopicPartitionCount;
+  protected long _minPartitionSampleStoreTopicRetentionTimeMs;
+  protected long _minBrokerSampleStoreTopicRetentionTimeMs;
+  protected volatile double _loadingProgress;
+  protected Producer<byte[], byte[]> _producer;
+  protected volatile boolean _shutdown = false;
+
+  public static final String PARTITION_METRIC_SAMPLE_STORE_TOPIC_CONFIG = "partition.metric.sample.store.topic";
+  public static final String BROKER_METRIC_SAMPLE_STORE_TOPIC_CONFIG = "broker.metric.sample.store.topic";
+  public static final String SAMPLE_STORE_TOPIC_REPLICATION_FACTOR_CONFIG = "sample.store.topic.replication.factor";
+  public static final String PARTITION_SAMPLE_STORE_TOPIC_PARTITION_COUNT_CONFIG = "partition.sample.store.topic.partition.count";
+  public static final String BROKER_SAMPLE_STORE_TOPIC_PARTITION_COUNT_CONFIG = "broker.sample.store.topic.partition.count";
+  public static final String MIN_PARTITION_SAMPLE_STORE_TOPIC_RETENTION_TIME_MS_CONFIG = "min.partition.sample.store.topic.retention.time.ms";
+  public static final String MIN_BROKER_SAMPLE_STORE_TOPIC_RETENTION_TIME_MS_CONFIG = "min.broker.sample.store.topic.retention.time.ms";
+
+  @Override
+  public void configure(Map<String, ?> config) {
+    _partitionMetricSampleStoreTopic = (String) config.get(PARTITION_METRIC_SAMPLE_STORE_TOPIC_CONFIG);
+    _brokerMetricSampleStoreTopic = (String) config.get(BROKER_METRIC_SAMPLE_STORE_TOPIC_CONFIG);
```

Comment:
I am confused. Why this this class use the same configs as `KafkaSampleStore`? Isn't this class supposed to generate samples that are collected during an ongoing execution to a new topic specific to partitions?